### PR TITLE
Include tests in Github Actions for JVM

### PR DESCRIPTION
**Pull request by laplab**
It turns out current commands in Github Actions for JVM do not include tests. We fix this by using `gradle build` instead of `gradle assemble`

There are no test tasks in assemble dry run:

<details>
  <summary>./gradlew assemble --dry-run</summary>

```
Starting a Gradle Daemon (subsequent builds will be faster)
:compileJava SKIPPED
:processResources SKIPPED
:classes SKIPPED
:jar SKIPPED
:assemble SKIPPED
:datatypes:compileKotlin SKIPPED
:datatypes:compileJava SKIPPED
:datatypes:processResources SKIPPED
:datatypes:classes SKIPPED
:datatypes:inspectClassesForKotlinIC SKIPPED
:datatypes:jar SKIPPED
:datasource:compileKotlin SKIPPED
:datasource:compileJava SKIPPED
:datasource:processResources SKIPPED
:datasource:classes SKIPPED
:datasource:inspectClassesForKotlinIC SKIPPED
:datasource:jar SKIPPED
:logical-plan:compileKotlin SKIPPED
:logical-plan:compileJava SKIPPED
:logical-plan:processResources SKIPPED
:logical-plan:classes SKIPPED
:logical-plan:inspectClassesForKotlinIC SKIPPED
:logical-plan:jar SKIPPED
:fuzzer:compileKotlin SKIPPED
:fuzzer:compileJava SKIPPED
:fuzzer:processResources SKIPPED
:fuzzer:classes SKIPPED
:fuzzer:inspectClassesForKotlinIC SKIPPED
:fuzzer:jar SKIPPED
:optimizer:compileKotlin SKIPPED
:optimizer:compileJava SKIPPED
:optimizer:processResources SKIPPED
:optimizer:classes SKIPPED
:optimizer:inspectClassesForKotlinIC SKIPPED
:optimizer:jar SKIPPED
:physical-plan:compileKotlin SKIPPED
:physical-plan:compileJava SKIPPED
:physical-plan:processResources SKIPPED
:physical-plan:classes SKIPPED
:physical-plan:inspectClassesForKotlinIC SKIPPED
:physical-plan:jar SKIPPED
:query-planner:compileKotlin SKIPPED
:query-planner:compileJava SKIPPED
:query-planner:processResources SKIPPED
:query-planner:classes SKIPPED
:query-planner:inspectClassesForKotlinIC SKIPPED
:query-planner:jar SKIPPED
:sql:compileKotlin SKIPPED
:sql:compileJava SKIPPED
:sql:processResources SKIPPED
:sql:classes SKIPPED
:sql:inspectClassesForKotlinIC SKIPPED
:sql:jar SKIPPED
:execution:compileKotlin SKIPPED
:execution:compileJava SKIPPED
:execution:processResources SKIPPED
:execution:classes SKIPPED
:execution:inspectClassesForKotlinIC SKIPPED
:execution:jar SKIPPED
:benchmarks:compileKotlin SKIPPED
:benchmarks:compileJava SKIPPED
:benchmarks:compileScala SKIPPED
:benchmarks:processResources SKIPPED
:benchmarks:classes SKIPPED
:benchmarks:inspectClassesForKotlinIC SKIPPED
:benchmarks:jar SKIPPED
:benchmarks:startScripts SKIPPED
:benchmarks:distTar SKIPPED
:benchmarks:distZip SKIPPED
:benchmarks:assemble SKIPPED
:protobuf:extractIncludeProto SKIPPED
:protobuf:extractProto SKIPPED
:protobuf:generateProto SKIPPED
:protobuf:compileKotlin SKIPPED
:protobuf:compileJava SKIPPED
:protobuf:processResources SKIPPED
:protobuf:classes SKIPPED
:protobuf:inspectClassesForKotlinIC SKIPPED
:protobuf:jar SKIPPED
:client:compileKotlin SKIPPED
:client:compileJava SKIPPED
:client:processResources SKIPPED
:client:classes SKIPPED
:client:inspectClassesForKotlinIC SKIPPED
:client:jar SKIPPED
:client:assemble SKIPPED
:datasource:assemble SKIPPED
:datatypes:assemble SKIPPED
:examples:compileKotlin SKIPPED
:examples:compileJava SKIPPED
:examples:compileScala SKIPPED
:examples:processResources SKIPPED
:examples:classes SKIPPED
:examples:inspectClassesForKotlinIC SKIPPED
:examples:jar SKIPPED
:examples:assemble SKIPPED
:execution:assemble SKIPPED
:executor:compileKotlin SKIPPED
:executor:compileJava SKIPPED
:executor:processResources SKIPPED
:executor:classes SKIPPED
:executor:inspectClassesForKotlinIC SKIPPED
:executor:jar SKIPPED
:executor:startScripts SKIPPED
:executor:distTar SKIPPED
:executor:distZip SKIPPED
:executor:assemble SKIPPED
:fuzzer:assemble SKIPPED
:jdbc:compileKotlin SKIPPED
:jdbc:compileJava SKIPPED
:jdbc:processResources SKIPPED
:jdbc:classes SKIPPED
:jdbc:inspectClassesForKotlinIC SKIPPED
:jdbc:jar SKIPPED
:jdbc:assemble SKIPPED
:logical-plan:assemble SKIPPED
:optimizer:assemble SKIPPED
:physical-plan:assemble SKIPPED
:protobuf:assemble SKIPPED
:query-planner:assemble SKIPPED
:sql:assemble SKIPPED

BUILD SUCCESSFUL in 16s
```

</details>

But test tasks are included in build dry run:

<details>
  <summary>./gradlew build --dry-run</summary>

```
:compileJava SKIPPED
:processResources SKIPPED
:classes SKIPPED
:jar SKIPPED
:assemble SKIPPED
:spotlessCheck SKIPPED
:compileTestJava SKIPPED
:processTestResources SKIPPED
:testClasses SKIPPED
:test SKIPPED
:check SKIPPED
:build SKIPPED
:datatypes:compileKotlin SKIPPED
:datatypes:compileJava SKIPPED
:datatypes:processResources SKIPPED
:datatypes:classes SKIPPED
:datatypes:inspectClassesForKotlinIC SKIPPED
:datatypes:jar SKIPPED
:datasource:compileKotlin SKIPPED
:datasource:compileJava SKIPPED
:datasource:processResources SKIPPED
:datasource:classes SKIPPED
:datasource:inspectClassesForKotlinIC SKIPPED
:datasource:jar SKIPPED
:logical-plan:compileKotlin SKIPPED
:logical-plan:compileJava SKIPPED
:logical-plan:processResources SKIPPED
:logical-plan:classes SKIPPED
:logical-plan:inspectClassesForKotlinIC SKIPPED
:logical-plan:jar SKIPPED
:fuzzer:compileKotlin SKIPPED
:fuzzer:compileJava SKIPPED
:fuzzer:processResources SKIPPED
:fuzzer:classes SKIPPED
:fuzzer:inspectClassesForKotlinIC SKIPPED
:fuzzer:jar SKIPPED
:optimizer:compileKotlin SKIPPED
:optimizer:compileJava SKIPPED
:optimizer:processResources SKIPPED
:optimizer:classes SKIPPED
:optimizer:inspectClassesForKotlinIC SKIPPED
:optimizer:jar SKIPPED
:physical-plan:compileKotlin SKIPPED
:physical-plan:compileJava SKIPPED
:physical-plan:processResources SKIPPED
:physical-plan:classes SKIPPED
:physical-plan:inspectClassesForKotlinIC SKIPPED
:physical-plan:jar SKIPPED
:query-planner:compileKotlin SKIPPED
:query-planner:compileJava SKIPPED
:query-planner:processResources SKIPPED
:query-planner:classes SKIPPED
:query-planner:inspectClassesForKotlinIC SKIPPED
:query-planner:jar SKIPPED
:sql:compileKotlin SKIPPED
:sql:compileJava SKIPPED
:sql:processResources SKIPPED
:sql:classes SKIPPED
:sql:inspectClassesForKotlinIC SKIPPED
:sql:jar SKIPPED
:execution:compileKotlin SKIPPED
:execution:compileJava SKIPPED
:execution:processResources SKIPPED
:execution:classes SKIPPED
:execution:inspectClassesForKotlinIC SKIPPED
:execution:jar SKIPPED
:benchmarks:compileKotlin SKIPPED
:benchmarks:compileJava SKIPPED
:benchmarks:compileScala SKIPPED
:benchmarks:processResources SKIPPED
:benchmarks:classes SKIPPED
:benchmarks:inspectClassesForKotlinIC SKIPPED
:benchmarks:jar SKIPPED
:benchmarks:startScripts SKIPPED
:benchmarks:distTar SKIPPED
:benchmarks:distZip SKIPPED
:benchmarks:assemble SKIPPED
:spotlessInternalRegisterDependencies SKIPPED
:benchmarks:spotlessKotlin SKIPPED
:benchmarks:spotlessKotlinCheck SKIPPED
:benchmarks:spotlessCheck SKIPPED
:benchmarks:compileTestKotlin SKIPPED
:benchmarks:compileTestJava SKIPPED
:benchmarks:compileTestScala SKIPPED
:benchmarks:processTestResources SKIPPED
:benchmarks:testClasses SKIPPED
:benchmarks:test SKIPPED
:benchmarks:check SKIPPED
:benchmarks:build SKIPPED
:protobuf:extractIncludeProto SKIPPED
:protobuf:extractProto SKIPPED
:protobuf:generateProto SKIPPED
:protobuf:compileKotlin SKIPPED
:protobuf:compileJava SKIPPED
:protobuf:processResources SKIPPED
:protobuf:classes SKIPPED
:protobuf:inspectClassesForKotlinIC SKIPPED
:protobuf:jar SKIPPED
:client:compileKotlin SKIPPED
:client:compileJava SKIPPED
:client:processResources SKIPPED
:client:classes SKIPPED
:client:inspectClassesForKotlinIC SKIPPED
:client:jar SKIPPED
:client:assemble SKIPPED
:client:spotlessKotlin SKIPPED
:client:spotlessKotlinCheck SKIPPED
:client:spotlessCheck SKIPPED
:client:compileTestKotlin SKIPPED
:client:compileTestJava SKIPPED
:client:processTestResources SKIPPED
:client:testClasses SKIPPED
:client:test SKIPPED
:client:check SKIPPED
:client:build SKIPPED
:datasource:assemble SKIPPED
:datasource:spotlessKotlin SKIPPED
:datasource:spotlessKotlinCheck SKIPPED
:datasource:spotlessCheck SKIPPED
:datasource:compileTestKotlin SKIPPED
:datasource:compileTestJava SKIPPED
:datasource:processTestResources SKIPPED
:datasource:testClasses SKIPPED
:datasource:test SKIPPED
:datasource:check SKIPPED
:datasource:build SKIPPED
:datatypes:assemble SKIPPED
:datatypes:spotlessKotlin SKIPPED
:datatypes:spotlessKotlinCheck SKIPPED
:datatypes:spotlessCheck SKIPPED
:datatypes:compileTestKotlin SKIPPED
:datatypes:compileTestJava SKIPPED
:datatypes:processTestResources SKIPPED
:datatypes:testClasses SKIPPED
:datatypes:test SKIPPED
:datatypes:check SKIPPED
:datatypes:build SKIPPED
:examples:compileKotlin SKIPPED
:examples:compileJava SKIPPED
:examples:compileScala SKIPPED
:examples:processResources SKIPPED
:examples:classes SKIPPED
:examples:inspectClassesForKotlinIC SKIPPED
:examples:jar SKIPPED
:examples:assemble SKIPPED
:examples:spotlessKotlin SKIPPED
:examples:spotlessKotlinCheck SKIPPED
:examples:spotlessCheck SKIPPED
:examples:compileTestKotlin SKIPPED
:examples:compileTestJava SKIPPED
:examples:compileTestScala SKIPPED
:examples:processTestResources SKIPPED
:examples:testClasses SKIPPED
:examples:test SKIPPED
:examples:check SKIPPED
:examples:build SKIPPED
:execution:assemble SKIPPED
:execution:spotlessKotlin SKIPPED
:execution:spotlessKotlinCheck SKIPPED
:execution:spotlessCheck SKIPPED
:execution:compileTestKotlin SKIPPED
:execution:compileTestJava SKIPPED
:execution:processTestResources SKIPPED
:execution:testClasses SKIPPED
:execution:test SKIPPED
:execution:check SKIPPED
:execution:build SKIPPED
:executor:compileKotlin SKIPPED
:executor:compileJava SKIPPED
:executor:processResources SKIPPED
:executor:classes SKIPPED
:executor:inspectClassesForKotlinIC SKIPPED
:executor:jar SKIPPED
:executor:startScripts SKIPPED
:executor:distTar SKIPPED
:executor:distZip SKIPPED
:executor:assemble SKIPPED
:executor:spotlessKotlin SKIPPED
:executor:spotlessKotlinCheck SKIPPED
:executor:spotlessCheck SKIPPED
:executor:compileTestKotlin SKIPPED
:executor:compileTestJava SKIPPED
:executor:processTestResources SKIPPED
:executor:testClasses SKIPPED
:executor:test SKIPPED
:executor:check SKIPPED
:executor:build SKIPPED
:fuzzer:assemble SKIPPED
:fuzzer:spotlessKotlin SKIPPED
:fuzzer:spotlessKotlinCheck SKIPPED
:fuzzer:spotlessCheck SKIPPED
:fuzzer:compileTestKotlin SKIPPED
:fuzzer:compileTestJava SKIPPED
:fuzzer:processTestResources SKIPPED
:fuzzer:testClasses SKIPPED
:fuzzer:test SKIPPED
:fuzzer:check SKIPPED
:fuzzer:build SKIPPED
:jdbc:compileKotlin SKIPPED
:jdbc:compileJava SKIPPED
:jdbc:processResources SKIPPED
:jdbc:classes SKIPPED
:jdbc:inspectClassesForKotlinIC SKIPPED
:jdbc:jar SKIPPED
:jdbc:assemble SKIPPED
:jdbc:spotlessKotlin SKIPPED
:jdbc:spotlessKotlinCheck SKIPPED
:jdbc:spotlessCheck SKIPPED
:jdbc:compileTestKotlin SKIPPED
:jdbc:compileTestJava SKIPPED
:jdbc:processTestResources SKIPPED
:jdbc:testClasses SKIPPED
:jdbc:test SKIPPED
:jdbc:check SKIPPED
:jdbc:build SKIPPED
:logical-plan:assemble SKIPPED
:logical-plan:spotlessKotlin SKIPPED
:logical-plan:spotlessKotlinCheck SKIPPED
:logical-plan:spotlessCheck SKIPPED
:logical-plan:compileTestKotlin SKIPPED
:logical-plan:compileTestJava SKIPPED
:logical-plan:processTestResources SKIPPED
:logical-plan:testClasses SKIPPED
:logical-plan:test SKIPPED
:logical-plan:check SKIPPED
:logical-plan:build SKIPPED
:optimizer:assemble SKIPPED
:optimizer:spotlessKotlin SKIPPED
:optimizer:spotlessKotlinCheck SKIPPED
:optimizer:spotlessCheck SKIPPED
:optimizer:compileTestKotlin SKIPPED
:optimizer:compileTestJava SKIPPED
:optimizer:processTestResources SKIPPED
:optimizer:testClasses SKIPPED
:optimizer:test SKIPPED
:optimizer:check SKIPPED
:optimizer:build SKIPPED
:physical-plan:assemble SKIPPED
:physical-plan:spotlessKotlin SKIPPED
:physical-plan:spotlessKotlinCheck SKIPPED
:physical-plan:spotlessCheck SKIPPED
:physical-plan:compileTestKotlin SKIPPED
:physical-plan:compileTestJava SKIPPED
:physical-plan:processTestResources SKIPPED
:physical-plan:testClasses SKIPPED
:physical-plan:test SKIPPED
:physical-plan:check SKIPPED
:physical-plan:build SKIPPED
:protobuf:assemble SKIPPED
:protobuf:spotlessKotlin SKIPPED
:protobuf:spotlessKotlinCheck SKIPPED
:protobuf:spotlessCheck SKIPPED
:protobuf:extractIncludeTestProto SKIPPED
:protobuf:extractTestProto SKIPPED
:protobuf:generateTestProto SKIPPED
:protobuf:compileTestKotlin SKIPPED
:protobuf:compileTestJava SKIPPED
:protobuf:processTestResources SKIPPED
:protobuf:testClasses SKIPPED
:protobuf:test SKIPPED
:protobuf:check SKIPPED
:protobuf:build SKIPPED
:query-planner:assemble SKIPPED
:query-planner:spotlessKotlin SKIPPED
:query-planner:spotlessKotlinCheck SKIPPED
:query-planner:spotlessCheck SKIPPED
:query-planner:compileTestKotlin SKIPPED
:query-planner:compileTestJava SKIPPED
:query-planner:processTestResources SKIPPED
:query-planner:testClasses SKIPPED
:query-planner:test SKIPPED
:query-planner:check SKIPPED
:query-planner:build SKIPPED
:sql:assemble SKIPPED
:sql:spotlessKotlin SKIPPED
:sql:spotlessKotlinCheck SKIPPED
:sql:spotlessCheck SKIPPED
:sql:compileTestKotlin SKIPPED
:sql:compileTestJava SKIPPED
:sql:processTestResources SKIPPED
:sql:testClasses SKIPPED
:sql:test SKIPPED
:sql:check SKIPPED
:sql:build SKIPPED

BUILD SUCCESSFUL in 1s
```

</details>

### CODE REVIEW

**Comment by laplab** [comment]
@andygrove Should we include linters in Github Actions?
Also it looks like currently JVM tests in master are broken because of arrow dependency.

File: .github/workflows/jvm.yml [hunk 1]
```diff
@@ -26,6 +26,6 @@ jobs:
       - name: Grant execute permission for gradlew
         run: chmod +x jvm/gradlew
       - name: Build and run JVM tests with Gradle
-        run: cd jvm && ./gradlew clean assemble publishToMavenLocal
+        run: cd jvm && ./gradlew clean assemble test
```
Comment (andygrove) [inline .github/workflows/jvm.yml:29]: The spark module depends on artifacts from the jvm module, so we'll need to run the `publishToMavenLocal` target in addition to the `test` target.

**Comment by andygrove** [comment]
@laplab I fixed the jvm test failures (rolled back arrow for now)

**Comment by laplab** [comment]
For some reason I do not see kotlin CSV tests in completed task, will investigate a little bit later

**Pull request merged**